tests/e2e/framework/kubernetes.go

```diff
@@ -97,39 +97,43 @@ func newKubeInfo(tmpDir, runID string) *KubeInfo {
 		Verbosity:  verbosity,
 		TmpDir:     tmpDir,
 		yamlDir:    filepath.Join(tmpDir, "yaml"),
-		Istioctl:   util.NewIstioctl(tmpDir, *namespace, *managerHub, *managerTag),
+		Istioctl:   NewIstioctl(tmpDir, *namespace, *managerHub, *managerTag),
 	}
 }
 
 // Setup set up Kubernetes prerequest for tests
 func (k *KubeInfo) Setup() error {
-	if err := os.Mkdir(k.yamlDir, os.ModeDir|os.ModePerm); err != nil {
+	var err error
+	if err = os.Mkdir(k.yamlDir, os.ModeDir|os.ModePerm); err != nil {
 		return err
 	}
-	if err := k.Istioctl.Install(); err != nil {
+
+	if err = k.Istioctl.Install(); err != nil {
```

Comment:
It's kind of tricky, since deploying app requires both istioctl and "ISTIO_MANAGER_ADDRESS", while on the other hand, setting up istioctl have to wait manager pod.  As a solution, I change apps to appManager and make it cleanable too. And move appDeploy to appManager.Setup()